batch-job.go

```diff
@@ -305,6 +306,73 @@ func (adm *AdminClient) GenerateBatchJob(_ context.Context, opts GenerateBatchJo
 	return "", fmt.Errorf("unknown batch job requested: %s", opts.Type)
 }
 
+// SupportedBatchJobsHeader is the header key for supported batch jobs included
+// by the server in the list batch jobs response.
+const SupportedBatchJobsHeader = "X-Minio-Supported-Batch-Jobs"
```

Comment:
why have this kind of an API? why not get a JSON response?

Reply:
It was to save an additional request (to list supported ones) when server reports that a job is not supported on generate, but it's probably cleaner to do a separate API. Will update.